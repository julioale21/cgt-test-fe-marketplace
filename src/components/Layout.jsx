import { Outlet } from 'react-router-dom';
import { Header } from '../components/shared/Header';
import { Box } from '@mui/material';

export function Layout() {
  const cartItemsCount = 3;

  return (
    <Box sx={{ display: 'flex', flexDirection: 'column', minHeight: '100vh' }}>
      <Header cartItemsCount={cartItemsCount} />

      <Box component="main" sx={{ flexGrow: 1, p: 3 }}>
        <Outlet />
      </Box>

      <Box component="footer" sx={{ p: 2, textAlign: 'center', mt: 'auto' }}>
        <p>Â© 2025</p>
      </Box>
    </Box>
  );
}
