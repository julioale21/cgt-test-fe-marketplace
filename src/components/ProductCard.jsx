import React from 'react';

import { Link } from 'react-router-dom';

import { Button, Card, CardActions, CardContent, CardMedia, Grid, Typography } from '@mui/material';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';

const ProductCard = ({ product }) => {
  return (
    <Grid item xs={12} sm={6} md={4} lg={3}>
      <Card sx={{ height: '100%', display: 'flex', flexDirection: 'column' }}>
        <CardMedia
          component="img"
          height="200"
          image={product.image}
          alt={product.name}
          sx={{
            objectFit: 'cover',
            height: '200px'
          }}
        />
        <CardContent sx={{ flexGrow: 1 }}>
          <Typography gutterBottom variant="h6" component="h2">
            {product.name}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mb: 2 }}>
            {product.description}
          </Typography>
          <Typography variant="h6" color="primary" sx={{ mt: 2 }}>
            ${product.price.toFixed(2)}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mt: 1 }}>
            Formats: {product.format.join(', ')}
          </Typography>
        </CardContent>
        <CardActions sx={{ justifyContent: 'space-between' }}>
          <Button component={Link} to={`/products/${product.id}`} size="small" color="primary">
            View Details
          </Button>
          <Button size="small" variant="contained" color="primary" startIcon={<ShoppingCartIcon />}>
            Add to Cart
          </Button>
        </CardActions>
      </Card>
    </Grid>
  );
};

export { ProductCard };
