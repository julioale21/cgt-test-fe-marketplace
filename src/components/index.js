export { ContainedButton } from "./buttons/ContainedButton";
export { Header } from "./shared/Header";
export { Layout } from "./Layout";
export { OutlinedButton } from "./buttons/OutlinedButton";
export { ProductList } from "./ProductList";
export {ProductCard} from "./ProductCard";
