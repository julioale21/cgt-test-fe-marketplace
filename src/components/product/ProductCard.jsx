import React from 'react';

import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
import { Card, CardActions, CardContent, CardMedia, Grid, Typography } from '@mui/material';
import { OutlinedButton } from '../buttons/OutlinedButton';
import { ContainedButton } from '../buttons/ContainedButton';



const ProductCard = ({ product }) => {
  if (!product) {
    return null;
  }

  const { id, name, price, description, image, format, rating, polygons, textures } = product;

  return (
    <Grid item xs={12} sm={6} md={4} lg={3}>
      <Card sx={{ height: '100%', display: 'flex', flexDirection: 'column' }}>
        <CardMedia
          component="img"
          height="200"
          image={image}
          alt={name}
          sx={{
            objectFit: 'cover',
            height: '200px'
          }}
        />
        <CardContent sx={{ flexGrow: 1 }}>
          <Typography gutterBottom variant="h6" component="h2">
            {name}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mb: 2 }}>
            {description}
          </Typography>
          <Typography variant="h6" color="primary" sx={{ mt: 2 }}>
            ${price.toFixed(2)}
          </Typography>
          <Typography variant="body2" color="text.secondary">
            Rating: {rating}/5
          </Typography>
          <Typography variant="body2" color="text.secondary">
            Polygons: {polygons}
          </Typography>
          <Typography variant="body2" color="text.secondary">
            Textures: {textures}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mt: 1 }}>
            Formats: {format.join(', ')}
          </Typography>
        </CardContent>
        <CardActions
          sx={{
            justifyContent: 'space-between',
            px: 1.5,
            pb: 1.5,
            pt: 0
          }}
        >
          <OutlinedButton component={Link} to={`/products/${id}`}>
            View Details
          </OutlinedButton>
          <ContainedButton startIcon={<ShoppingCartIcon sx={{ fontSize: '1rem' }} />}>
            Add to Cart
          </ContainedButton>
        </CardActions>
      </Card>
    </Grid>
  );
};

ProductCard.propTypes = {
  product: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    format: PropTypes.arrayOf(PropTypes.string).isRequired,
    polygons: PropTypes.string.isRequired,
    textures: PropTypes.string.isRequired
  })
};

export { ProductCard };
