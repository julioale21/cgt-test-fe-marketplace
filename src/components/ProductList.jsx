import React from 'react';
import { Grid } from '@mui/material';
import { ProductCard } from './ProductCard';

const ProductList = ({ products }) => {
  return (
    <Grid container spacing={4}>
      {products.map((product) => (
        <ProductCard key={product.id} product={product} />
      ))}
    </Grid>
  );
};

export { ProductList };
