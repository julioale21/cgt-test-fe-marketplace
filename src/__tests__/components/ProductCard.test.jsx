// ProductCard.test.jsx
import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { ProductCard } from '../../components/ProductCard';

const mockProduct = {
  id: 1,
  name: '3D Modern Chair',
  price: 149.99,
  description: 'High-quality 3D model of a modern minimalist chair',
  image: '/path/to/image.jpg',
  category: 'Furniture',
  rating: 4.5,
  format: ['FBX', 'OBJ', 'MAX'],
  polygons: '24k',
  textures: '4K PBR'
};

const renderProductCard = (props = {}) => {
  return render(
    <BrowserRouter>
      <ProductCard product={mockProduct} {...props} />
    </BrowserRouter>
  );
};

describe('ProductCard', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <ProductCard product={mockProduct} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot when no product is provided', () => {
    const { container } = render(<ProductCard product={null} />);
    expect(container).toMatchSnapshot();
  });

  it('renders product information correctly', () => {
    renderProductCard();

    expect(screen.getByText(mockProduct.name)).toBeInTheDocument();
    expect(screen.getByText(mockProduct.description)).toBeInTheDocument();
    expect(screen.getByText(`$${mockProduct.price.toFixed(2)}`)).toBeInTheDocument();
    expect(screen.getByText(`Rating: ${mockProduct.rating}/5`)).toBeInTheDocument();
    expect(screen.getByText(`Polygons: ${mockProduct.polygons}`)).toBeInTheDocument();
    expect(screen.getByText(`Textures: ${mockProduct.textures}`)).toBeInTheDocument();
    expect(screen.getByText(`Formats: ${mockProduct.format.join(', ')}`)).toBeInTheDocument();
  });

  it('renders product image with correct attributes', () => {
    renderProductCard();

    const image = screen.getByRole('img');
    expect(image).toHaveAttribute('src', mockProduct.image);
    expect(image).toHaveAttribute('alt', mockProduct.name);
  });

  it('renders action buttons', () => {
    renderProductCard();

    expect(screen.getByText('View Details')).toBeInTheDocument();
    expect(screen.getByText('Add to Cart')).toBeInTheDocument();
  });

  it('has correct link to product details', () => {
    renderProductCard();

    const detailsLink = screen.getByText('View Details').closest('a');
    expect(detailsLink).toHaveAttribute('href', `/products/${mockProduct.id}`);
  });

  it('returns null when no product is provided', () => {
    const { container } = render(<ProductCard product={null} />);
    expect(container).toBeEmptyDOMElement();
  });
});
