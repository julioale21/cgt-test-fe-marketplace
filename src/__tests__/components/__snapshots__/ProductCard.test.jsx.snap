// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard matches snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-em32m3-MuiPaper-root-MuiCard-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <img
          alt="3D Modern Chair"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-qapfwi-MuiCardMedia-root"
          height="200"
          src="/path/to/image.jpg"
        />
        <div
          class="MuiCardContent-root css-x4lmbv-MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1wmifpn-MuiTypography-root"
          >
            3D Modern Chair
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2 css-19b5ozw-MuiTypography-root"
          >
            High-quality 3D model of a modern minimalist chair
          </p>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-11ary1m-MuiTypography-root"
          >
            $
            149.99
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1wle3ir-MuiTypography-root"
          >
            Rating: 
            4.5
            /5
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1wle3ir-MuiTypography-root"
          >
            Polygons: 
            24k
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1wle3ir-MuiTypography-root"
          >
            Textures: 
            4K PBR
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-tmy7vb-MuiTypography-root"
          >
            Formats: 
            FBX, OBJ, MAX
          </p>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-ksj18d-MuiCardActions-root"
        >
          <a
            href="/products/1"
          >
            View Details
          </a>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-uwf23-MuiButtonBase-root-MuiButton-root"
            id=":r1:"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wdc28j-MuiSvgIcon-root"
                data-testid="ShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2M1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2"
                />
              </svg>
            </span>
            Add to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProductCard matches snapshot when no product is provided 1`] = `<div />`;
